src/QueryM.purs

```diff
@@ -114,20 +132,61 @@ utxosAt' addr = do
     affFunc :: (Either Error JsonWsp.UtxoQR -> Effect Unit) -> Effect Canceler
     affFunc cont = do
       let
-        ls = listeners config.ws
-        ws = underlyingWebSocket config.ws
+        ls = listeners config.ogmiosWs
+        ws = underlyingWebSocket config.ogmiosWs
       ls.utxo.addMessageListener id
         ( \result -> do
             ls.utxo.removeMessageListener id
-            (allowError cont) $ result
+            allowError cont $ result
         )
       _wsSend ws sBody
       pure $ Canceler $ \err -> do
         liftEffect $ ls.utxo.removeMessageListener id
         liftEffect $ throwError $ err
   liftAff $ makeAff $ affFunc
 
-allowError :: (Either Error JsonWsp.UtxoQR -> Effect Unit) -> JsonWsp.UtxoQR -> Effect Unit
+--------------------------------------------------------------------------------
+-- Datum Cache Queries
+--------------------------------------------------------------------------------
+
+getDatumByHash :: DatumHash -> QueryM (Maybe PlutusData)
+getDatumByHash hash = do
+  queryDatumCache (GetDatumByHashRequest hash) >>= case _ of
+    GetDatumByHashResponse mData -> pure $ mData
+    _ -> liftEffect $ throw "Request-response type mismatch. Should not have happened"
```

Comment:
Since this effect is already returning a `Maybe PlutusData`, why not return a `Nothing` here since the caller already needs to deal with the `Maybe`?